internal/controller/accessrequest_controller.go

```diff
@@ -156,6 +162,46 @@ func (r *AccessRequestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return result, nil
 }
 
+type AccessRequestConflictError struct {
+	message string
+}
+
+func (e *AccessRequestConflictError) Error() string {
+	return e.message
+}
+
+func NewAccessRequestConflictError(msg string) *AccessRequestConflictError {
+	return &AccessRequestConflictError{
+		message: msg,
+	}
+}
+
+// Validate will verify if there are existing AccessRequests for the same
+// user/app/role already in progress.
+func (r *AccessRequestReconciler) Validate(ctx context.Context, ar *api.AccessRequest) error {
```

Comment:
```suggestion
func (r *AccessRequestReconciler) ValidateNewAccessRequest(ctx context.Context, ar *api.AccessRequest) error {
```

Would this be more accurate?

Reply:
Actually, there is no guarantee that the controller will always reconcile new AccessRequests. When the code reach that point, we could have situations where it is an existing AccessRequest.

Reply:
Sure... so we could theoretically hit a situation where two get applied before reconciliation and then whichever we pick up first "wins."

Of course, with the `arResp.Status.RequestState == ""` check I requested, I think we might actually just deadlock. Both will fail validation forever.

Reply:
yeah.. if the controller is down and 2 access requests are created for the same user/app/role, one of them should proceed and the other be marked as invalid. Reverted `arResp.Status.RequestState == ""` check.